### PR TITLE
RelOptRulesTest improvements

### PR DESCRIPTION
**Pull request by rubenada**
Some refactoring and clean-up on RelOptRulesTest following [this suggestion](https://issues.apache.org/jira/browse/CALCITE-4621?focusedCommentId=17352671&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17352671) by @julianhyde ,

### CODE REVIEW

**Comment by vlsi** [comment]
What do you think if `Function<RelBuilder, RelNode>` was replaced with`Consumer<RelBuilder>` and `build()` was called by the calling code?

**Review by liyafan82 — APPROVED** [review]
Looks like sound simplifications.

**Comment by julianhyde** [comment]
Conventionally when defining a lambda for `relFn` we call the `RelBuilder` `b`, not `relBuilder`. It makes the code more concise when we use the `RelBuilder` multiple times to create expressions. Can you stick to that convention.

I think the `relFn` parameter and field need a `@Nullable` annotation.

Why are there so many renames of "planBefore" to "planAfter" and vice versa? I can't tell whether these are real changes or just instability in the file format. If you can, preserve the file format. Generally "planBefore" should occur before "planAfter" (even though that is not alphabetical).

**Comment by rubenada** [comment]
> Conventionally when defining a lambda for relFn we call the RelBuilder b, not relBuilder. It makes the code more concise when we use the RelBuilder multiple times to create expressions. Can you stick to that convention.

Done.

> I think the relFn parameter and field need a @Nullable annotation.

Annotation added.

> Why are there so many renames of "planBefore" to "planAfter" and vice versa? I can't tell whether these are real changes or just instability in the file format. If you can, preserve the file format. Generally "planBefore" should occur before "planAfter" (even though that is not alphabetical).

This is because I have homogenize several tests. Originally, some of the tests were checking "manually" a certain plan with some boilerplate code e.g.
```
RelNode output = hepPlanner.findBestExp();
final String planAfter = NL + RelOptUtil.toString(output);
final DiffRepository diffRepos = getDiffRepos();
diffRepos.assertEquals("planAfter", "${planAfter}", planAfter);
SqlToRelTestBase.assertValid(output);
```

This has been refactored to reuse `RelOptTestBase#Sql` methods:
- `check`: for the plans that must change when the rules are applied. They require `${planBefore}` and `${planAfter}`, and they must be different.
- `checkUnchanged`: for the plans that must not change when rules are applied. They require only `${planBefore}` (which is used to times, before and after the rules application.

Originally, some tests were checking an unchanged plan by checking manually the after plan, and they only defined `${planAfter}` in the xml file. With the proposed homogenization, these tests are now using `checkUnchanged` (and the plan is defined as `${planBefore}`). 
Analogously, some tests were checking a changed plan by manually checking the after plan, and they only defined `${planAfter}` in the xml file. These tests are now using `check`, and they define also their corresponding `${planBefore}` too.

To sum up, we can have tests (the unchanged ones) that define only `${planBefore}`; and tests (the changed ones) that defines  `${planBefore}` and `${planAfter}` (which must be different). I agree that the appropriate order for the latter is `${planBefore}` and then `${planAfter}`, I have tried to apply this convention; @julianhyde do you have any concrete example where this order is inversed in the patch?

**Comment by rubenada** [comment]
Are there any other comments about this PR? Otherwise I plan to squash commits and merge in the coming days.
